[quant][graphmode][fx] Fix fp16 dynamic quant for functional linear

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52414 [qunat][graphmode][fx] Enable test for non quantized input for cat
* #52413 [quant][graphmode][fx] Guard the supported quantization type for add/mul
* #52412 [quant][graphmode][fx] Enable test for non quantized input for add/mul
* **#52369 [quant][graphmode][fx] Fix fp16 dynamic quant for functional linear**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26491425](https://our.internmc.facebook.com/intern/diff/D26491425)

jerryzh168 committed:
[quant][graphmode][fx] Fix fp16 dynamic quant for functional linear

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Fix fp16 dynamic quant for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Fix fp16 dynamic quant for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26491425](https://our.internmc.facebook.com/intern/diff/D26491425)

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantize_fx.py:
```diff
@@ -457,7 +457,7 @@ def forward(self, x):
         tests = [
             (Linear, (linear_weight,), (linear_input,),
-             ns.call_function(torch.ops.quantized.linear_dynamic),
+             ns.call_function(torch.ops.quantized.linear_dynamic_fp16),
```
Do we need to update any other tests? I see few others with float16_dynamic_qconfig

jerryzh168 replied:
checked the tests, looks like we only need to update this one

supriyar reviewed (approved):
lgtm, please check all tests in test_quantize_fx that use fp16 qconfig before landing

jerryzh168 committed:
Update on "[quant][graphmode][fx] Fix fp16 dynamic quant for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26491425](https://our.internmc.facebook.com/intern/diff/D26491425)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Fix fp16 dynamic quant for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26491425](https://our.internmc.facebook.com/intern/diff/D26491425)

[ghstack-poisoned]
